osf/utils/notifications.py

```diff
@@ -5,10 +5,6 @@
 from osf.utils.workflows import RegistrationModerationTriggers
 
 def get_email_template_context(resource):
-    provider = getattr(resource, 'provider', False)
-    assert provider, 'This resource does not fit the template'
```

Comment:
Is it possible/likely that this will be called with resources that don't have a provider? If so, we want to do these checks instead of letting things die with an AttributeError, we should just use more expressive exceptions than `assert` outside of test code. If it's not likely/currently possible, then this is fine.